### DISCUSSION TITLE
[QUESTION] Local & External Models

### DISCUSSION THREAD

**Discussion by martineliascz** [open-webui/open-webui#4433] [General]
Dear Open WebUI Team,

I've been exploring the Open WebUI and have come across the "Local Models" and "External Models" features. I'm quite curious about these and would greatly appreciate some clarification. Could you please explain what these terms refer to and how exactly they differ in functionality? Additionally, I'm interested in understanding if there are any specific scenarios where one might be preferred over the other.

Thank you in advance for your assistance.

Martin

![screenshot](https://github.com/user-attachments/assets/83ec272e-2ef6-4ce9-9887-f724417af47b)

**Comment by raydoomed** [comment]
我也想知道。。。搞不清楚这是什么

**Comment by avviano** [comment]
Local model are models you have on your hard drive. External ones are via API. That's my understanding.

**Comment by turicas** [comment]
As far as I could tell from the source code, every model carries a `connection_type` field that is either "local" or "external":
- Ollama models default to local ([ollama.py#L361](https://github.com/open-webui/open-webui/blob/8dae237a0bfdac4b7f55b463b3e2769ea4b94a0b/backend/open_webui/routers/ollama.py#L361))
- OpenAI-compatible API models default to external ([openai.py#L439](https://github.com/open-webui/open-webui/blob/8dae237a0bfdac4b7f55b463b3e2769ea4b94a0b/backend/open_webui/routers/openai.py#L439))

The two dropdowns map to the config keys `TASK_MODEL` and `TASK_MODEL_EXTERNAL` ([config.py#L1809-L1819](https://github.com/open-webui/open-webui/blob/8dae237a0bfdac4b7f55b463b3e2769ea4b94a0b/backend/open_webui/config.py#L1809-L1819)). The selection logic is in the `get_task_model_id` function at [utils/task.py#L18-L29](https://github.com/open-webui/open-webui/blob/8dae237a0bfdac4b7f55b463b3e2769ea4b94a0b/backend/open_webui/utils/task.py#L18-L29):
- If chatting with a local model -> the Local Task Model is used for background tasks
- If chatting with an external model -> the External Task Model is used
- If either dropdown is empty -> it falls back to the chat model itself (which can be costly if you're using a frontier model)

I use all models via external API and set only the external task model to `gpt-5.4-nano`, since tasks like title generation are non-critical.
